pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java

```diff
@@ -838,7 +840,12 @@ public int getControllerBrokerPortOverride() {
   }
 
   public String getControllerResourcePackages() {
-    return getProperty(CONTROLLER_RESOURCE_PACKAGES, DEFAULT_CONTROLLER_RESOURCE_PACKAGES);
+    Object rawProperty = getRawProperty(CONTROLLER_RESOURCE_PACKAGES, DEFAULT_CONTROLLER_RESOURCE_PACKAGES);
```

Comment:
I think we should fix `getProperty(String name, String defaultValue)` instead of handling it here. `getProperty(String name, String defaultValue)` should return comma separated string instead of `List.toString()`.